### PR TITLE
fix(xref/script): some exceptions are interfaces

### PR DESCRIPTION
**Pull request by marcoscaceres**
Also refactored howToCite by splitting it in two... this makes it a bit easier to maintain, hopefully.

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Something occurred to me when working on this regarding enums... the following could totally work too:

```
{{ EnumContainer/"some enum value" }} 
```

Which is the same as: 

```
{{EnumContainer["some enum value"]}}
```

We should support both forms. That's less Editors need to learn.

**Comment by marcoscaceres** [comment]
Will file a ReSpec bug...

File: static/xref/script.js [hunk 1]
```diff
@@ -117,35 +124,47 @@ function renderResults(entries, query) {
...
-        <td>${howToCite}</td>
+        <td>${cite}</td>
       </tr>`;
     html += row;
   }
   output.innerHTML = html;
 }
 
+function howToCiteIDL(term, entry) {
+  const { type, for: forList } = entry;
```
Comment (sidvishnoi) [inline static/xref/script.js]: Nit: Lets keep using `forContext` for consistency (or `forContextList` maybe)

File: static/xref/script.js [hunk 2]
```diff
@@ -93,21 +93,28 @@ function getFormData() {
 }
 
 async function handleSubmit() {
+  if (form.querySelector("input[name='term']").value === '') return;
```
Comment (sidvishnoi) [inline static/xref/script.js]: Nit: Use `if (data.term === "") return;`

Comment (sidvishnoi) [reply static/xref/script.js]: Also, lets make `input[name='term']` as required in HTML.

File: static/xref/script.js (hunk 1 above)
Comment (sidvishnoi) [reply static/xref/script.js]: Maybe na.

**Comment by marcoscaceres** [comment]
Thanks for fixing the nits!

**Pull request merged**